### ISSUE TITLE
Event details links all lead to 404

### ISSUE THREAD

**Issue by tdentremont** [Automattic/eventbrite-api#74]
I have the eventbrite-index.php template customized for my theme and working fine, but the single event template does not work. All links from the index lead to 404 errors. 
I have declared the necessary theme support. Is there an extra permalink setting not mentioned? 
Others have experienced the same as commented in the support forum on WordPress.org.

**Comment by kwight** [comment]
Do you have an example of a 404 URL? Meaning, is there anything in the format of the URL that gives a hint as to the problem?

**Comment by tdentremont** [comment]
I can't give a publically facing example as I'm developing this on my local system. The event titles in the archive page seem to be pointing to single posts for each of the events, but it seems like one of the endpoints isn't working. I figured if this plugin hasn't been updated in some time and is no longer being supported by Automattic, I might just hack the plugin itself and remove the conditional statement that replaces the eventbrite URL for the event, with the single post's permalink, so that the URLs are only pointing to the eventbrite URL and not worry about a single post.

**Comment by madebyelmcity** [comment]
@tdentremont Did you get this working? I'm having the same issue.

Everything worked fine with a test account. Now that I've got our "real" account connected, I'm getting 404s too.

**Comment by tdentremont** [comment]
Hi @madebyelmcity , no I haven't done that yet, but I plan to do as kwight suggested. I might just get the master branch of the repo and merge it locally with the branch commited by EstherGodoy in her EB Plugin Enhancements V2 #71. If that doesn't work I may just take another approach entirely and leave out connecting to EB altogether. It turns out that EventBrite has their own widgets for this purpose, but those are only available to premium account holders.

**Comment by kwight** [comment]
> Everything worked fine with a test account. Now that I've got our "real" account connected, I'm getting 404s too.

Ooooh, that sounds suspiciously like https://github.com/Automattic/eventbrite-api/issues/32

**Comment by tdentremont** [comment]
Yeah, this thing is frustrating me to no end. I think I'll just create a new custom post type and have the client simply enter the minimum data to display and link to the event on EB. This way I can also add a calendar to the mix and be done with this plugin. Too bad. I guess every time EB makes changes to their API, the plugin devs need to make changes. Just like facebook.......

**Comment by kwight** [comment]
@tdentremont were you able to determine if your problem was related to the 32bit issue mentioned above, or something else?

Yes, it's been frustrating to keep things up to pace with Eventbrite's API (their v3 transition was tough with many landmines and surprises). It's at the point where the plugin needs a solid suite of tests added, and an overhaul – in the meantime, #71 is the closest thing to a most recent version. Like all projects, it comes down to time and priorities, so I can't give any sort of ETA unfortunately.

> It turns out that EventBrite has their own widgets for this purpose, but those are only available to premium account holders.

The widgets are quite decent, as long as you don't need any customization, nor care about their styling matching the theme design (IIRC they're all iframes, meaning you're stuck with their styling).

> I think I'll just create a new custom post type and have the client simply enter the minimum data to display and link to the event on EB.

Yep, great option; if you don't need theme-wide integrations or custom tools, this plugin can be overkill (although it's nice it handles all of the OAuth headaches).

**Comment by tdentremont** [comment]
@kwight The 32-bit issue is not the case. I'm using a 64-bit system myself. I think for the sake of saving time I'll just opt to creating the aforementioned custom post type. I prefer being able to style the archive and single posts myself, without worry that something might break down the road. Better safe than sorry. If, this plugin gets a major overhaul and pleases the masses, I'll take another stab at it.

**Comment by madebyelmcity** [comment]
FWIW, using #71 worked fine for me.

**Comment by mmjaeger** [comment]
If I add the "Event" page as a sub-page of e.g. "Company" I'll get those 404 errors pointing to the details page - however if I create the Event page with no parent, the links seem to work - I guess it has something to do with the Url rewriting (permalink) - hope somebody can come up with a fix.

**Comment by kwight** [comment]
@mmjaeger you're right, and it's a known issue that the Event page has to be a parent page. We have no ETA for a fix at this point. Have you tried PR #71 ? It has many improvements and fixes that seem to help various issues.

**Comment by mmjaeger** [comment]
@kwight - tried to use the code mentioned in #71 - as soon as I put in an organizer_id it doesn't show anything.

**Comment by sadikyalcin** [comment]
It it obviously broken. For those who are still looking for this by any chance, I've created 3 gists showing how you can implement your own index/single views for your events;

1. Register query vars and add rewrite rule for loading the template in your themes `functions.php` as shown [here](https://gist.github.com/sadikyalcin/7c5ff7fe71cae959bf4a0fe07ab89600)
2. Add custom link to events on [index page](https://gist.github.com/sadikyalcin/4a2f614ab68a01e7c2a0c136f0bb7d5c)
3. Load events by id on [single view](https://gist.github.com/sadikyalcin/c0e6500e99f7a7509faa7e3c0438df1a)
